iceoryx_utils/cmake/IceoryxPlatform.cmake

```diff
@@ -58,6 +58,8 @@ function(iox_create_asan_compile_time_blacklist BLACKLIST_FILE_PATH)
         file(WRITE  ${BLACKLIST_FILE_PATH} "# This file is auto-generated from iceoryx_utils/cmake/IceoryxPlatform.cmake\n")
         file(APPEND ${BLACKLIST_FILE_PATH} "# src:*file_name.cpp*\n")
         file(APPEND ${BLACKLIST_FILE_PATH} "# fun:*Test_Name*\n")
+        file(APPEND ${BLACKLIST_FILE_PATH} "# posh_runtime.cpp:70:12: runtime error: reference binding to null pointer of type 'iox::runtime::PoshRuntime'\n")
+        file(APPEND ${BLACKLIST_FILE_PATH} "src:*posh_runtime.cpp\n")
```

Comment:
Do you have more information what fails? Maybe we can fix it with this PR

Reply:
I would keep that separate because it does not belong into this issue: https://github.com/eclipse-iceoryx/iceoryx/issues/423

Reply:
you're right, but maybe it's a minor thing which could be solved within a few minutes.

Reply:
Here you [go](https://github.com/prasannabhat/iceoryx/runs/1585366319?check_suite_focus=true)
(Dont know how long these build logs are persisted)

Reply:
That's really weird. The error doesn't make sense. I tried to reproduce it on my machine, but I just got the `reference binding to null pointer` error.
I fixed that and removed the runtime from the blacklist. Let's see what happens

Reply:
> Let's see what happens
... well, it's broken :D

@dkroenke the tests for setting the runtime factory are broken. We need a RouDi for this tests, but cannot use the RouDiEnvironment, since it sets the factory itself and when we override that in the test, we break the RouDiEnvironment. I'm wondering why it doesn't crash.

I disabled the test with a note to re-enable them with #449

Reply:
okay, windows is broken and I have no idea why :/

Reply:
it builds now ... but I have only slightly an idea why :D